examples/04_fuzzy_joining.py

```diff
@@ -392,33 +392,24 @@
 
 # We will test four possible values of match_score:
 params = {
-    "joiner-1__match_score": [0.2, 0.9],
-    "joiner-2__match_score": [0.2, 0.9],
-    "joiner-3__match_score": [0.2, 0.9],
+    "joiner-1__match_score": [0.1, 0.9],
+    "joiner-2__match_score": [0.1, 0.9],
+    "joiner-3__match_score": [0.1, 0.9],
 }
 
-grid = GridSearchCV(pipeline, param_grid=params)
+grid = GridSearchCV(pipeline, param_grid=params, cv=cv)
 grid.fit(df, y)
 
-print(grid.best_params_)
+print("Best parameters:", grid.best_params_)
 
-##########################################################################
-# The grid searching gave us the best value of 0.5 for the parameter
-# ``match_score``. Let's use this value in our regression:
-#
-
-print(f"Mean R2 score with pipeline is {grid.score(df, y):.2f}")
```

Comment:
@Vincent-Maladiere having a closer look, this part does not make sense: we are scoring on the training data. if we cross-validate the grid search correctly the score does not improve

Reply:
depending on the `cv` random state the grid search score can be the same or worse than the one without grid search

Reply:
Are you improving the results while doing the grid search? If yes, I think this should be kept.